crates/aiken-lang/src/uplc.rs

```diff
@@ -2566,10 +2560,19 @@ impl<'a> CodeGenerator<'a> {
                 });
             }
 
+            assert_vec.push(Air::Trace {
+                scope: scope.clone(),
+                tipo: tipo.clone(),
+            });
+
+            assert_vec.push(Air::String {
+                scope: scope.clone(),
+                value: "Constr index did not match any type variant".to_string(),
+            });
+
             assert_vec.push(Air::ErrorTerm {
                 scope,
                 tipo: tipo.clone(),
-                label: Some("Constr index did not match any type variant".to_string()),
```

Comment:
@rvcas This is the only thing I am truly unsure about. This PR slightly rework the AIR too so that `ErrorTerm` is only a term. This one is a bit special because it was an error term directly pushed from within the code generation. So I am trying to mimic the new behavior by replacing the label with two other steps `Trace` and `String` before that.

Reply:
Nevermind, tested that with the following:

```
type Foo {
  F1
  F2
}

type Bar {
  B1
  B2
  B3
}

test foo() {
  let x: Bar = B3
  let x: Data = x

  expect y: Foo = x

  y == F1
}
```

And it triggers the error fine with the right trace warning 👌